src/Game/Game.js

```diff
@@ -1,28 +1,160 @@
-import {useState} from "react";
+import { useState, useEffect } from "react";
 import Board from "./board";
-import {NumOfSquares, boardRows, boardCols} from "../Utils/utility"
+import { numOfSquares, boardCols, boardRows } from "../Utils/utility";
 
 const Game = () => {
-    const [grid, setGrid] = useState(Array(NumOfSquares).fill(null)) //fills an array the size of the game board with null
-    // const [head, setHead] = useState()
+  //initial direction
+  const [direction, setDirection] = useState("movingRight");
+  //fills an array the size of the game board with null
+  const [grid, setGrid] = useState(Array(numOfSquares).fill(null));
+  //array to keep track of snake length
+  const [snake, setSnake] = useState(Array(1).fill(null));
 
-    // const snakeHead = () => {
+  const [score, setScore] = useState(3);
 
-    // }
+  // Everything needed for game start
+  function startGame() {
+    if (grid.indexOf(5) === -1) {
+      const startGrid = [...grid];
+      appleSpawn(startGrid);
+      startGrid[0] = 5;
+      setGrid(startGrid);
+    } else {
+      alert("Game Already Started");
+    }
+  }
 
-    // function handleClick() {
-    //     console.log(NumOfSquares)
-    //     console.log(boardCols)
-    //     console.log(boardRows)
-    //     console.log(width)
-    //     console.log(height)
-    // }
+  //restores default states
+  const reset = () => {
+    setGrid(Array(numOfSquares).fill(null));
+    setSnake(Array(1).fill(null));
+    setScore(3);
+    setDirection("movingRight");
+  };
+
+
+  function endConditions(fromIndex, toIndex) {
+    for (var i = 0; i < boardRows; i++) {
+      if (
+        (fromIndex === i * boardCols - 1 && toIndex === i * boardCols) || // crosses edge going right
+        (fromIndex === i * boardCols && toIndex === i * boardCols - 1) || // crosses edge going left
+        toIndex > numOfSquares - 1 || // crosses edge going down
+        grid[toIndex] === 3 || // crosses snake body
+        toIndex < 0 // crosses edge going up
+      ) {
+        alert("Game Over!");
+        reset();
+        return true;
+      }
+    }
+  }
+
+  // spawns apple in an unoccupied spot
+  function appleSpawn(array) {
+    const randomSpawn = Math.floor(Math.random() * numOfSquares);
+    if (array[randomSpawn] != null) appleSpawn(array)
+    else array[randomSpawn] = 0;
+  }
+
+  // checks to see if an apple has been eaten
+  const appleEaten = () => {
+    if (grid.indexOf(0) === -1) {
+      appleSpawn(grid);
+      const newScore = score + 3;
+      setScore(newScore);
+      console.log("Apple Eaten! Your New Score Is ", newScore);
+    } else return;
+  };
+
+  // maintains snake length based on score 
+  function maintainLength() {
+    if (snake.length < score) return;
+    else {
+      const targetIndex = snake[0];
+      snake.shift();
```

Comment:
You shouldn't directly modify state variables, this is an anti pattern. You should setSnake(updatedSnake)